src/components/panel/TextViewsToggle.tsx

```diff
@@ -65,7 +67,7 @@ const TextViewsToggle: FC = () => {
                     onClick={(e) => handleTextViewClick(e, button.view)}
                     variant={panelState.view === button.view ? 'secondary' : 'ghost'}
                     size="icon"
-                    disabled={!panelState.contentTypes?.length}
+                    disabled={!panelState.contentTypes?.length || (!panelState.existsImage && ['image', 'swap', 'split'].includes(button.view))}
```

Comment:
Move this to a function `isEnabled`, reverse the condition, we should always stay positive :)